monitoring/mock_uss/geoawareness/parsers/ed269/__init__.py

```diff
@@ -0,0 +1,151 @@
+import json
+from enum import Enum
+from typing import List, Any, Optional, Union, Dict, Tuple
+from implicitdict import ImplicitDict, StringBasedDateTime, StringBasedTimeDelta
+
+
+class Restriction(str, Enum):
+    PROHIBITED = "PROHIBITED"
+    REQ_AUTHORISATION = "REQ_AUTHORISATION"
+    CONDITIONAL = "CONDITIONAL"
+    NO_RESTRICTION = "NO_RESTRICTION"
+
+
+class Reason(str, Enum):
+    AIR_TRAFFIC = "AIR_TRAFFIC"
+    SENSITIVE = "SENSITIVE"
+    PRIVACY = "PRIVACY"
+    POPULATION = "POPULATION"
+    NATURE = "NATURE"
+    NOISE = "NOISE"
+    FOREIGN_TERRITORY = "FOREIGN_TERRITORY"
+    EMERGENCY = "EMERGENCY"
+    OTHER = "OTHER"
+
+
+class YESNO(str, Enum):
+    YES = "YES"
+    NO = "NO"
+
+
+class Purpose(str, Enum):
+    AUTHORIZATION = "AUTHORIZATION"
+    NOTIFICATION = "NOTIFICATION"
+    INFORMATION = "INFORMATION"
+
+
+class UASZoneAuthority(ImplicitDict):
+    name: Optional[str]  # max length: 200
+    service: Optional[str]  # max length: 200
+    email: Optional[str]
+    contactName: Optional[str]  # max length: 200
+    siteURL: Optional[str]
+    phone: Optional[str]  # max length: 200
+    purpose: Optional[Purpose]
+    intervalBefore: Optional[str]
+
+
+class VerticalReferenceType(str, Enum):
+    AGL = "AGL"
+    AMSL = "AMSL"
+
+
+# TODO: Discuss with @ben about Union[PolygonType, CircleType] which throws a not implemented exception
```

Comment:
Ug, implicit polymorphism is difficult :(  I don't think it's a particularly solvable problem in general (how would one tell if one was receiving an A or a B in general when the declared type was Union[A, B]?) and really it's probably not worth addressing in an automatic framework.  For the purpose of deserialization as part of a complex ImplicitDict type, the escape valve is to just declare the field a `dict`.  That way, there won't be any schema checking as part of ImplicitDict deserialization, but the data will still be available as a plain dict which can be checked manually.  ImplicitDict is just a fancy dict, so it never hurts to revert to a plain dict -- all that is lost is some automatic validation features and syntactic sugar.

TL;DR: Instead of `my_field: Union[PolygonType, CircleType]`, one possibility is to just do `my_field: dict`

Reply:
The fallback to simple dict is good to know, thanks :)